client/my-sites/sidebar-unified/index.jsx

```diff
@@ -50,6 +57,37 @@ export const MySitesSidebarUnified = ( { path } ) => {
 		return <Spinner className="sidebar-unified__menu-loading" />;
 	}
 
+	// Checks if there is a Happy Chat active and user clicks on an External link.
+	// On which case we show a modal awaiting for user confirmation for opening that
+	// link on new tab in order to avoid Happy Chat session disconnection.
+	// We return a bool that shows if the logic should terminate here.
+	const continueInCalypso = ( url, event ) => {
+		if ( isHappychatSessionActive && isExternal( url ) ) {
+			event && event.preventDefault();
+			setExternalUrl( url );
+			setShowDialog( true );
+			dispatch(
+				recordTracksEvent( 'calypso_nav_unification_external_link_dialog_show', {
+					link: url,
+				} )
+			);
+			return false;
+		}
+		return true;
+	};
+
+	const closeModalHandler = ( openUrl ) => {
+		setShowDialog( false );
+		if ( openUrl ) {
+			openUrl && window.open( externalUrl );
```

Comment:
Nitpick: you no longer need to check `openUrl` since you're already doing it one line above:
```suggestion
		window.open( externalUrl );
```

Reply:
wow, I missed that! Thanks!